### PR TITLE
reload containerd on config changes

### PR DESCRIPTION
**Pull request by aojea**
Wait until the system finish to boot to restart containerd, after we patch it
Fixes: https://github.com/kubernetes-sigs/kind/issues/2262

### CODE REVIEW

**Review by BenTheElder — CHANGES_REQUESTED** [review]
the way we're doing image loading is considered a bug to fix.
we should not mimic it here. the tools are available to do this correctly.

**Comment by aojea** [comment]
TIL `systemctl is-system-running --wait`

**Comment by aojea** [comment]
fedora images seems to not be available
```

URL: ["https://vagrantcloud.com/fedora/33-cloud-base"]
Error: The requested URL returned error: 403
Bringing machine 'default' up with 'virtualbox' provider..
```

**Comment by AkihiroSuda** [comment]
> @AkihiroSuda should we switch to fedora34?

I have a PR to switch the default to 34: https://github.com/kubernetes-sigs/kind/pull/2220
But depends on https://github.com/kubernetes-sigs/kind/pull/2230.

Opened https://github.com/kubernetes-sigs/kind/pull/2279 as an attempt to deflake 33.

File: pkg/cluster/internal/create/actions/config/config.go [hunk 1]
```diff
@@ -185,8 +185,8 @@ func (a *Action) Execute(ctx *actions.ActionContext) error {
 					return errors.Wrap(err, "failed to write patched containerd config")
 				}
 				// restart containerd now that we've re-configured it
-				// skip if the systemd (also the containerd) is not running
-				if err := node.Command("bash", "-c", `! systemctl is-system-running || systemctl restart containerd`).Run(); err != nil {
+				// wait until boot has finished
```
Comment (BenTheElder) [inline pkg/cluster/internal/create/actions/config/config.go:188]: we want to only restart it if we observed it was already running after completing the patching, if it wasn't running waiting and restarting it is:
a) slow
b) unnecessary (as it never read the old unpatched config)

there's still a race as to wether or not it starts before we check to restart it, but if we check to restart it and it wasn't running we're fine and if it was running we restart it and wether or not it read the old config it's fine and we don't waste much time.

the problem previously is an ineffective check for it it is running.

**Comment by BenTheElder** [comment]
filed https://github.com/kubernetes-sigs/kind/pull/2382 https://github.com/kubernetes-sigs/kind/pull/2274#discussion_r675212707

**Comment by aojea** [comment]
Duplicate

**Pull request closed without merging**